src/Hooks/Dispatchers/ReplaceCommandHookDispatcher.php

```diff
@@ -17,6 +18,15 @@ class ReplaceCommandHookDispatcher extends HookDispatcher
 {
 
     /**
+     * @var LoggerInterface|null
+     */
+    protected $logger = null;
+
+    public function setLogger(LoggerInterface $logger) {
```

Comment:
Should use LoggerAwareInterface / LoggerAwareTrait. The later is an optimization (avoid duplicate code), but the former is critical, as it will trigger dependency injection for classes in the DI container that implement it.